### PR TITLE
Update versioneer to version 0.29

### PR DESCRIPTION
**Pull request by speleo3**
Installed versioneer 0.29 with Python 3.12 and ran

```sh
versioneer install
```

Fixes the following error when doing `pip install .` with Python 3.12:
```
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [18 lines of output]
      propka/versioneer.py:421: SyntaxWarning: invalid escape sequence '\s'
        LONG_VERSION_PY['git'] = '''
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "propka/setup.py", line 10, in <module>
          version=versioneer.get_version(),
                  ^^^^^^^^^^^^^^^^^^^^^^^^
        File "propka/versioneer.py", line 1480, in get_version
          return get_versions()["version"]
                 ^^^^^^^^^^^^^^
        File "propka/versioneer.py", line 1412, in get_versions
          cfg = get_config_from_root(root)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "propka/versioneer.py", line 344, in get_config_from_root
          parser.readfp(f)
          ^^^^^^^^^^^^^
      AttributeError: 'ConfigParser' object has no attribute 'readfp'. Did you mean: 'read'?
      [end of output]

```

### CODE REVIEW

**Review by IAlibay — COMMENTED** [review]
Would it be worth also bumping up https://github.com/jensengroup/propka/blob/8ba1692e42ec504a75a5f9a08223b25bb8f9d024/.github/workflows/python-package.yml#L21 to include 3.12?

**Review by orbeckst — APPROVED** [review]
lgtm – I might have to update elsewhere, too

**Comment by speleo3** [comment]
> Would it be worth also bumping up to include 3.12?

Yes, good catch.

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.8", "3.9", "3.10", "3.11"]
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
```
Comment (IAlibay) [inline .github/workflows/python-package.yml:21]: 🙈 sorry I also forgot that setup.py's classifiers also need updating

**Pull request merged**